backend/src/db/providers/testProvider.ts

```diff
@@ -88,4 +89,21 @@ const getCourseTest = async (courseId: string | ObjectId): Promise<ITest> => {
   return test as unknown as ITest;
 };
 
-export { getTestProvider, getTestById, getTrueAnswersProvider, updateTestQuestions, getCourseTest };
+const addCourseTest = async (testData: ITest) => {
+  const properQuestionsToSet = setAnswerProperNumbersToQuestions(testData.questions);
+
+  return TestModel.create({
+    title: testData.title,
+    questions: properQuestionsToSet,
+    timeout: testData.timeout,
+  });
+};
```

Comment:
You should only create test here. All the other actions should be done outside provider. Move setting proper numbers to questions outside provider, pls